src/webapi/rendering_context.rs

```diff
@@ -27,7 +183,771 @@ impl RenderingContext for CanvasRenderingContext2d {
+        js_try! ( @(no_return)
+            @{&self.0}.addColorStop(@{offset}, @{color});
+        ).unwrap()
+    }
+}
+
+impl ImageData {
+
+    /// Returns a Uint8ClampedArray representing a one-dimensional array containing the data in the RGBA order, 
+    /// with integer values between 0 and 255 (included).
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/ImageData/data)
+    pub fn get_data(&self) -> Vec<u8> {
+        js! (
+            return @{&self.0}.data;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the number of rows in the image data object.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/ImageData/height)
+    pub fn get_height(&self) -> f64 {
+        js! (
+            return @{&self.0}.height;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the number of pixels per row in the image data object.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/ImageData/width)
+    pub fn get_width(&self) -> f64 {
+        js! (
+            return @{&self.0}.width;
+        ).try_into().unwrap()
+    }
+}
+
 impl CanvasRenderingContext2d {
+    /// # Properties
+    
+    /// The CanvasRenderingContext2D.canvas property is a read-only reference to the HTMLCanvasElement 
+    /// object that is associated with the context. It might be null if there is no association with an <canvas> element.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/canvas)
+    pub fn get_canvas(&self) -> Option<CanvasElement> {
+        js! (
+            @{&self.0}.canvas;
+        ).try_into().unwrap()
+    }
+
+    /// The CanvasRenderingContext2D.fillStyle property of the Canvas 2D API specifies the color or style to use inside shapes. 
+    /// The default is #000 (black).
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/fillStyle)
+    pub fn get_fill_style(&self) -> String {
+        js! (
+            @{&self.0}.fillStyle
+        ).try_into().unwrap()
+    }
+
+    /// The CanvasRenderingContext2D.fillStyle property of the Canvas 2D API specifies the color or style to use inside shapes. 
+    /// The default is #000 (black).
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/fillStyle)
+    pub fn set_fill_style_color(&self, color: &str){
+        js! { @(no_return)
+            @{&self.0}.fillStyle = @{color};
+        }
+    }
+
+    /// The CanvasRenderingContext2D.fillStyle property of the Canvas 2D API specifies the color or style to use inside shapes. 
+    /// The default is #000 (black).
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/fillStyle)
+    pub fn set_fill_style_gradient(&self, gradient: CanvasGradient){
```

Comment:
In cases like this it'd be better to take `&CanvasGradient` instead of `CanvasGradient`. (Otherwise the user will have to `clone()` the argument every time they call this function.)